### PR TITLE
Update makepattern.py for python3 and modern nxo64

### PR DESCRIPTION
**Pull request by averne**

### CODE REVIEW

**Comment by averne** [comment]
Ok it's broken, will reopen once it works

**Comment by averne** [comment]
Well it was actually working, I was looking at the wrong file... Sorry about that

File: pattern/makepattern.py [hunk 1]
```diff
@@ -45,9 +46,9 @@ def main(filenames):
...
 							mask[i] &= ~(0xFFF << 10)
 
-			mask_str = ''.join(struct.pack('<I', m) for m in mask)
-			bits = ''.join(struct.pack('<I', i&m) for i,m in zip(instrs, mask))
-			regex = ''.join((re.escape(i) if m == '\xFF' else r'[\0-\xff]') for i, m in zip(bits, mask_str))
+			mask_str = b''.join(struct.pack('<I', m) for m in mask)
+			bits = b''.join(struct.pack('<I', i&m) for i,m in zip(instrs, mask))
+			regex = b''.join((re.escape(i.to_bytes(1)) if m == 0xff else b'[\0-\xff]') for i, m in zip(bits, mask_str))
 
 			positions = [m.start() for m in re.finditer(regex, textstr)]
```
Comment (TSRBerry) [inline pattern/makepattern.py:53]: With these changes python2 is stuck at this line.
I waited for ~12 minutes and the output file was still empty, so I stopped the script there.

File: pattern/makepattern.py [hunk 2]
```diff
@@ -45,9 +46,9 @@ def main(filenames):
 						if arm64.get_Rn(instr) in adrp_regs:
 							mask[i] &= ~(0xFFF << 10)
 
-			mask_str = ''.join(struct.pack('<I', m) for m in mask)
-			bits = ''.join(struct.pack('<I', i&m) for i,m in zip(instrs, mask))
-			regex = ''.join((re.escape(i) if m == '\xFF' else r'[\0-\xff]') for i, m in zip(bits, mask_str))
+			mask_str = b''.join(struct.pack('<I', m) for m in mask)
+			bits = b''.join(struct.pack('<I', i&m) for i,m in zip(instrs, mask))
+			regex = b''.join((re.escape(i.to_bytes(1)) if m == 0xff else b'[\0-\xff]') for i, m in zip(bits, mask_str))
```
Comment (TSRBerry) [inline pattern/makepattern.py:51]: I wasn't able to find a way that works for both py2 and 3 without issues, so let's just use the old version for py2 and the new version for py3:

```suggestion
			if sys.version_info[0] > 2:
			    mask_str = b''.join(struct.pack('<I', m) for m in mask)
				bits = b''.join(struct.pack('<I', i & m) for i, m in zip(instrs, mask))
				regex = b''.join((re.escape(i.to_bytes(1)) if m == 0xff else b'[\0-\xff]') for i, m in zip(bits, mask_str))
			else:
				mask_str = ''.join(struct.pack('<I', m) for m in mask)
				bits = ''.join(struct.pack('<I', i & m) for i, m in zip(instrs, mask))
				regex = ''.join((re.escape(i) if m == '\xFF' else r'[\0-\xff]') for i, m in zip(bits, mask_str))
```

**Review by TSRBerry — CHANGES_REQUESTED** [review]
Thank you very much, I haven't worked on this in quite some time!

As a test I ran the old version (with python2) against the ACNH sdk of the base game and compared the output with the new version (with python3 and 2).

I noticed the output is a lot smaller when using python3. It still contains everything, but the regexes (?) look a lot better.

Is your `applypattern-ghidra.py` script able to handle these as well?

**Comment by averne** [comment]
> Is your `applypattern-ghidra.py` script able to handle these as well?

Yes, as far as I understand (didn't look too deep), it's because the py3 produces a bytestring vs a normal string with py2, with has to get escaped (so with a lot more backward slashes)

File: pattern/makepattern.py (hunk 1 above)
Comment (averne) [reply pattern/makepattern.py:58]: Thanks for the suggestion, I didn't realize you were trying to keep those scripts compatible with python2.

Comment (TSRBerry) [reply pattern/makepattern.py:58]: Just in case SciresM modifies his version of these again or wants to use this one. :D

**Review by TSRBerry — APPROVED** [review]
Lgtm now! Thank you again! :D

**Pull request merged**